src/methods/iterator.js

```diff
@@ -177,12 +269,18 @@ export function IteratorClose(realm: Realm, iterator: ObjectValue, completion: A
   if (ret instanceof UndefinedValue) return completion;
 
   // 5. Let innerResult be Call(return, iterator, « »).
-  let innerResult = Call(realm, ret.throwIfNotConcrete(), iterator, []);
+  let innerResult;
+  try {
+    innerResult = Call(realm, ret.throwIfNotConcrete(), iterator, []);
+  } catch (error) {
+    innerResult = error;
+  }
```

Comment:
Doing that here results in some failing tests. Here the instruction doesn’t put a `?` before `Call`, and in the next step its specified that if `completion` is an error it should be thrown instead of `innerResult`.

One of the failing tests is [`language/expressions/assignment/dstr-array-rest-iter-thrw-close-err.js`](https://github.com/tc39/test262/blob/2a0da2efd40306653269bb53564924a656402c7f/test/language/expressions/assignment/dstr-array-rest-iter-thrw-close-err.js). Here there are two errors which may be thrown. `ReturnError` may be thrown by the `return` method which is called by this line in `IteratorClose`, and `Test262Error` which is thrown in the destructure pattern on [line 64](https://github.com/tc39/test262/blob/2a0da2efd40306653269bb53564924a656402c7f/test/language/expressions/assignment/dstr-array-rest-iter-thrw-close-err.js#L64). Both will be thrown, but the test expects that it will be the `Test262Error` that “wins.” We call `IteratorClose` after [`thrower()`](https://github.com/tc39/test262/blob/2a0da2efd40306653269bb53564924a656402c7f/test/language/expressions/assignment/dstr-array-rest-iter-thrw-close-err.js#L64) throws the `Test262Error`, as a “finally” type thing I guess. The `Test262Error` is represented as the `completion` argument in `IteratorClose`. If we do not catch the error thrown by `Call` (which will be the `ReturnError` in this case) then the `Test262Error` will be wrongly ignored instead of the `ReturnError`. This test confused me when I first saw it, does that explanation make sense?

Perhaps we should modify the code for step 7 so that it throws? It returns in master and I didn’t change it.